src/modules/auth/store/user/user.types.ts

```diff
@@ -1,6 +1,7 @@
-export type UserSchema = Record<string, unknown> | null;
+import { UserSchema } from '@auth/types';
 
 export interface UserState {
-	// TODO: add user typings from proxy
-	user: UserSchema;
+	user: UserSchema | null;
+	loading: boolean;
+	error: unknown | null;
```

Comment:
Waarom `UserSchema | null` over `user?: userSchema`

Reply:
Omdat de user property altijd in de store gaat aanwezig zijn met die optional lijkt het of die soms niet aanwezig is, voor niet ingevulde waardes gebruik ik dan liever `null` ipv `undefined`